src/pybind/mgr/dashboard/controllers/mgr_modules.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from . import ApiController, RESTController, Endpoint, UpdatePermission
+from .. import mgr
+from ..security import Scope
+from ..services.ceph_service import CephService
+from ..services.exception import handle_send_command_error
+from ..tools import find_object_in_list, str_to_bool
+
+
+@ApiController('/mgr/module', Scope.CONFIG_OPT)
+class MgrModules(RESTController):
+    managed_modules = ['telemetry']
+
+    def list(self):
+        """
+        Get the list of managed modules.
+        :return: A list of objects with the fields 'name' and 'enabled'.
+        :rtype: list
+        """
+        result = []
+        mgr_map = mgr.get('mgr_map')
+        for module_config in mgr_map['available_modules']:
+            if module_config['name'] in self.managed_modules:
+                result.append({'name': module_config['name'], 'enabled': False})
+        for name in mgr_map['modules']:
+            if name in self.managed_modules:
+                obj = find_object_in_list('name', name, result)
+                obj['enabled'] = True
+        return result
```

Comment:
Just a hint: Have you tried [list comprehensions](https://www.python.org/dev/peps/pep-0202/)? For some reason I find them easier to understand than step-by-step composition:
```python
mgr_map = mgr.get('mgr_map')
return [{
  'name': module_config['name'],
  'enabled': bool(find_object_in_list('name', module_config['name'], mgr_map['modules']))}
    for module_config in mgr_map['available_modules']
      if module_config['name'] in self.managed_modules]
```

Reply:
That's possible, but IMHO this code is unreadable. I'm wrong?